### PR TITLE
JKNS-257: updates around new multi container sample pod templates in the openshift jenkins image

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @rolfedh

@adambkaplan @akram also review if you have cycles 

- Aligned team: Dev Tools
- For branches: 4.10+
- Jira: https://issues.redhat.com/browse/JKNS-257
- Direct link to doc preview: https://deploy-preview-39610--osdocs.netlify.app/openshift-enterprise/latest/openshift_images/using_images/images-other-jenkins
- SME review: @gabemontero 
- QE review: @akram
- Peer review: @rolfedh 
- <When all reviews are complete change this string to "Ready for merge.">

### CODE REVIEW

**Comment by gabemontero** [comment]
FYI according to his slack status @rolfedh is out sick

in the interim @akram @adambkaplan if you all have cycles PTAL

**Comment by gabemontero** [comment]
> @gabemontero, please update the PR description, above, with the following boilerplate and fill in the placeholders:
> 
> ```
> - Aligned team: Dev Tools
> - For branches: <specify the release branch, such as "4.10+">
> - Jira: https://issues.redhat.com/browse/JKNS-257
> - Direct link to doc preview: https://deploy-preview-39610--osdocs.netlify.app/openshift-enterprise/latest/openshift_images/using_images/images-other-jenkins
> - SME review: @gabemontero 
> - QE review: @akram
> - Peer review: @rolfedh 
> - <When all reviews are complete change this string to "Ready for merge.">
> ```

description update via copy/paste

File: modules/images-other-jenkins-config-kubernetes.adoc [hunk 1]
```diff
@@ -76,6 +76,68 @@ data:
     </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
 ----
 
+Another example that uses multiple containers would be (where we use {product-title} ImageStream image references to
```
Comment (rolfedh) [inline modules/images-other-jenkins-config-kubernetes.adoc:79]: Please make this paragraph a single line.
```suggestion
The following example also uses multiple containers. It uses {product-title} `ImageStream` image references to refer to `ImageStreams` objects that are running in the 'openshift' namespace:```

File: modules/images-other-jenkins-config-kubernetes.adoc [hunk 2]
```diff
@@ -76,6 +76,68 @@ data:
...
+ImageStreams installed in the 'openshift' namespace that work when running in an {product-title} cluster):
```
Comment (rolfedh) [inline modules/images-other-jenkins-config-kubernetes.adoc:80]:
```suggestion
```

File: modules/images-other-jenkins-create-service.adoc [hunk 3]
```diff
@@ -38,3 +38,14 @@ $ oc new-app jenkins-persistent
 ----
 $ oc new-app jenkins-ephemeral
 ----
+
+With both templates, you can run 'oc describe' on them to see all the parameters available for overriding
+various settings for the resulting Jenkins deployment, including several of the xref:images-other-jenkins-env-var.adoc[environment variables] available
```
Comment (rolfedh) [inline modules/images-other-jenkins-create-service.adoc:43]:
```suggestion
```

File: modules/images-other-jenkins-create-service.adoc [hunk 4]
```diff
@@ -38,3 +38,14 @@ $ oc new-app jenkins-persistent
...
+on the image.
```
Comment (rolfedh) [inline modules/images-other-jenkins-create-service.adoc:44]:
```suggestion
```

**Comment by gabemontero** [comment]
suggestions pulled in, commits squashed @rolfedh

File: modules/images-other-jenkins-kubernetes-plugin.adoc [hunk 5]
```diff
@@ -100,3 +100,52 @@ spec:
 <9> The node stanza references the name of the defined pod template.
 
 By default, the pod is deleted when the build completes. This behavior can be modified with the plug-in or within a pipeline Jenkinsfile.
+
+Upstream Jenkins has more recently introduced a YAML declarative format for defining
+PodTemplates in-line with your pipelines.  An example of this format, using the sample 'java-builder'
```
Comment (rolfedh) [inline modules/images-other-jenkins-kubernetes-plugin.adoc:105]:
```suggestion
`PodTemplates` in-line with your pipelines.  An example of this format, using the sample 'java-builder'
```

File: openshift_images/using_images/images-other-jenkins-agent.adoc [hunk 6]
```diff
@@ -26,6 +26,22 @@ The Maven and Node.js Jenkins agent images provide Dockerfiles for the Universal
 Use and extend an appropriate agent image version for the your of {product-title}. If the `oc` client version that is embedded in the agent image is not compatible with the {product-title} version, unexpected behavior can result.
 ====
 
+{product-title} also provides sample PodTemplates defined in the {product-title}
+Jenkins image that illustrates how to use these agent images with the
+Jenkins Kubernetes plug-in.
```
Comment (rolfedh) [inline openshift_images/using_images/images-other-jenkins-agent.adoc:31]: Please remove all hard breaks from all paragraphs.

File: openshift_images/using_images/images-other-jenkins-agent.adoc [hunk 7]
```diff
@@ -26,6 +26,22 @@ The Maven and Node.js Jenkins agent images provide Dockerfiles for the Universal
...
+
+The four PodTemplates defined are:
```
Comment (rolfedh) [inline openshift_images/using_images/images-other-jenkins-agent.adoc:33]: Please replace all instances of "PodTemplates" with "`PodTemplates` <object type>" OR "pod templates".

File: openshift_images/using_images/images-other-jenkins-agent.adoc [hunk 8]
```diff
@@ -26,6 +26,22 @@ The Maven and Node.js Jenkins agent images provide Dockerfiles for the Universal
...
+- The 'maven' PodTemplate which uses a single container that uses the {product-title} Maven Jenkins agent image
```
Comment (rolfedh) [inline openshift_images/using_images/images-other-jenkins-agent.adoc:34]:
```suggestion
- The `maven` pod template, which uses a single container that uses the {product-title} Maven Jenkins agent image.
```
Please replace all single quotes with backticks.

**Review by rolfedh — APPROVED** [review]
Left some suggestions. Pre-approving.

**Comment by gabemontero** [comment]
latest round pulled in / process / squashed / repushed @rolfedh 

PTAL

File: modules/images-other-jenkins-config-kubernetes.adoc [hunk 9]
```diff
@@ -76,6 +76,67 @@ data:
     </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
 ----
 
+The following example also uses multiple containers. It uses {product-title} `ImageStream` image references to image streams that are running in the 'openshift' namespace:```
```
Comment (rolfedh) [inline modules/images-other-jenkins-config-kubernetes.adoc:79]:
```suggestion
The following example also uses multiple containers. It uses {product-title} `ImageStream` image references to image streams that are running in the `openshift` namespace:
```

File: modules/images-other-jenkins-create-service.adoc [hunk 10]
```diff
@@ -38,3 +38,12 @@ $ oc new-app jenkins-persistent
 ----
 $ oc new-app jenkins-ephemeral
 ----
+
+With both templates, you can run 'oc describe' on them to see all the parameters available for overriding.
```
Comment (rolfedh) [inline modules/images-other-jenkins-create-service.adoc:42]:
```suggestion
With both templates, you can run `oc describe` on them to see all the parameters available for overriding.
```

File: modules/images-other-jenkins-env-var.adoc [hunk 11]
```diff
@@ -90,4 +90,26 @@ log file to persist when a fatal error occurs. The fatal error file is saved at
...
+
+|`AGENT_BASE_IMAGE`
+|Setting this value overrides the image used for the 'jnlp' container in the sample kubernetes plug-in PodTemplates
+provided with this image.  Otherwise, the image from the 'jenkins-agent-base:latest' ImageStreamTag in the 'openshift'
+namespace is used.
+|Default:
+`image-registry.openshift-image-registry.svc:5000/openshift/jenkins-agent-base:latest`
+
+|`JAVA_BUILDER_IMAGE`
+|Setting this value overrides the image used for the 'java-builder' container in the 'java-builder' sample kubernetes
```
Comment (rolfedh) [inline modules/images-other-jenkins-env-var.adoc:102]: Please replace all instances of straight quotes (') with backticks (`) throughout.

File: modules/images-other-jenkins-env-var.adoc [hunk 12]
```diff
@@ -90,4 +90,26 @@ log file to persist when a fatal error occurs. The fatal error file is saved at
...
+plug-in PodTemplates provided with this image.  Otherwise, the image from the 'java:latest' ImageStreamTag in the
+'openshift' namespace is used.
+|Default:
+`image-registry.openshift-image-registry.svc:5000/openshift/java:latest`
+
+|`NODEJS_BUILDER_IMAGE`
+|Setting this value overrides the image used for the 'nodejs-builder' container in the 'nodejs-builder' sample kubernetes
+plug-in PodTemplates provided with this image.  Otherwise, the image from the 'nodejs:latest' ImageStreamTag in the
```
Comment (rolfedh) [inline modules/images-other-jenkins-env-var.adoc:110]: Please update PodTemplates throughout.

File: openshift_images/using_images/images-other-jenkins-agent.adoc [hunk 13]
```diff
@@ -26,6 +26,20 @@ The Maven and Node.js Jenkins agent images provide Dockerfiles for the Universal
 Use and extend an appropriate agent image version for the your of {product-title}. If the `oc` client version that is embedded in the agent image is not compatible with the {product-title} version, unexpected behavior can result.
 ====
 
+{product-title} also provides sample pod templates defined in the {product-title} Jenkins image that illustrates how to use these agent images with the Jenkins Kubernetes plug-in.
+
+The four pod templates defined are:
+- The `maven` pod template, which uses a single container that uses the {product-title} Maven Jenkins agent image.
+- The 'nodejs' pod template which uses a single container that uses the {product-title} Node.js Jenkins agent image.
+- The 'java-builder' pod template which employs 2 containers.  One is the 'jnlp' container which uses the {product-title}
```
Comment (rolfedh) [inline openshift_images/using_images/images-other-jenkins-agent.adoc:34]:
```suggestion
- The 'java-builder' pod template which employs two containers. One is the 'jnlp' container which uses the {product-title}
```

**Comment by gabemontero** [comment]
per discussion with @rolfedh in https://coreos.slack.com/archives/C014MHHKUSF/p1639421562133600 he is going to pick my commit here into his own PR and complete the final clean up for this.

I'll close this PR out when his PR is available.

**Comment by rolfedh** [comment]
@gabemontero  you can close this PR. I've created https://github.com/openshift/openshift-docs/pull/39873 instead (WIP).

**Pull request closed without merging**